### ISSUE TITLE
Cannot install Vim with the `--with-custom-ruby` option anymore

### ISSUE THREAD

**Issue by orenyk** [Homebrew/homebrew-core#22596]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [ ] ~~Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?~~ N/A (formula works but not as expected)
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
```
# brew config
HOMEBREW_VERSION: 1.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 987805d3dcd4c8dbc70127399d4ea7aee8e98c21
Last commit: 14 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 68ad718233b77b8672745d2763f8e318632309af
Core tap last commit: 12 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_NO_ANALYTICS_THIS_RUN: 1
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.15.1 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12.6-x86_64
Xcode: 9.2
CLT: 9.2.0.0.1.1510905681
X11: N/A

# brew doctor
Your system is ready to brew.
```

To help us debug your issue please explain:
- What you were trying to do (and why)
I'm trying to install Vim using the system-installed Ruby instead of the Homebrew `ruby` package to avoid conflicts with rbenv (discussion [here](https://apple.stackexchange.com/a/255889)). I was formerly able to use the option `--with-custom-ruby` when installing the Vim package that automatically used the system Ruby, but that option no longer seems to be present (it was apparently removed in 6b8cb0a several days ago)
- What happened (include command output)
```
$ brew install vim --with-custom-ruby
==> Installing dependencies for vim: perl, readline, libyaml, ruby, sqlite, gdbm, python
```
- What you expected to happen
Vim to be installed without also installing the Homebrew `ruby` package.
- Step-by-step reproduction instructions (by running `brew install` commands)
```
# As of 2018/01/06
brew update
brew install vim --with-custom-ruby
```

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

**Comment by ilovezfs** [comment]
Correct. This option has been removed and is no longer supported.

**Comment by dramalho** [comment]
Grand, but this is ruining `rbenv` like so

```shell
brew install vim
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (caskroom/cask, homebrew/core).
==> Updated Formulae
immortal

==> Installing dependencies for vim: ruby
==> Installing vim dependency: ruby
==> Downloading https://homebrew.bintray.com/bottles/ruby-2.5.0.sierra.bottle.tar.gz
Already downloaded: /Users/dramalho/Library/Caches/Homebrew/ruby-2.5.0.sierra.bottle.tar.gz

(yada yada yada)

$ which bundle
/usr/local/bin/bundle

$ ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16] <--- via rbenv

$ less /usr/local/bin/bundler
#!/usr/local/Cellar/ruby/2.5.0/bin/ruby  
# frozen_string_literal: true
```

if I remove ruby - and vim for the sake of respecting the dependencies - 

$ brew uninstall vim ruby
Uninstalling /usr/local/Cellar/vim/8.0.1400_4... (1,423 files, 22.8MB)
Uninstalling /usr/local/Cellar/ruby/2.5.0... (3,322 files, 18.4MB)

$ which bundle
/Users/dramalho/.rbenv/shims/bundle
```

and all is good 

It's a bit annoying, but most of all this broke things in an unexpected way and it would be great if the formula could be aware of rvm/rbenv - or ruby being present in general - and not force the installation  of ruby.

**Comment by ilovezfs** [comment]
You can move `/Users/dramalho/.rbenv/shims` earlier in your PATH.

**Comment by dramalho** [comment]
hummm it should be there already

```shell
export PATH="/Users/dramalho/.rbenv/shims:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:$PATH"
```

and I don't touch the path after this . Let me reinstall vim again

**Comment by ilovezfs** [comment]
`which bundle` should definitely point to the shim if it's earlier in the PATH. You can check
```
hash -r
which bundle
which -a bundle
```

**Comment by dramalho** [comment]
funny enough, I repeated the installation (vim, with ruby, etc) but the path to bundle didn't change, it's still pointing to the shims folder. I can confirm I haven't changed the path or anything in that regard.

Really really weird, but this is ok now. Thanks @ilovezfs , I'll keep this in mind, if it happens again or to my colleagues I'll try to get more information on top of this, but something definitely happened after I upgraded brew yesterday and the `ruby` installation seems - maybe - related.

Thanks again
